handlers/save_handler.py

```diff
@@ -0,0 +1,70 @@
+import re
+
+import requests
+
+from bottypes.command_descriptor import *
+import handlers.handler_factory as handler_factory
+from handlers.base_handler import *
+from util.savelinkhelper import unfurl
+
+GITHUB_REPO = ""  # owner/repo-name
+GITHUB_BRANCH = ""  # e.g master
+
+CATEGORIES = ["web", "pwn", "re", "crypto", "misc"]
+
+
+class SaveCommand(Command):
+    """Save a url from a slack message according to a specific category"""
+
+    @classmethod
+    def execute(cls, slack_wrapper, args, timestamp, channel_id, user_id, user_is_admin):
+        """Execute the save link command"""
+        if not GITHUB_REPO and not GITHUB_BRANCH:
+            raise InvalidCommand("Link saver not configured")
+        if args[0] not in CATEGORIES:
+            raise InvalidCommand("Invalid Category")
+
+        message = slack_wrapper.get_message(channel_id, timestamp)["messages"][0]["text"]
+        url_regex = "((https?):\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?"
+        url = re.search(url_regex, message)
+
+        if not url:
+            slack_wrapper.post_message(channel_id, "Couldn't extract URL", timestamp)
```

Comment:
Look at how errors are posted as replies (in a thread) this is cleaner.

Reply:
So, something like "Error: `unable to extract URL`" ?